WalletWasabi.Gui/Controls/NoparaPasswordBox.cs

```diff
@@ -0,0 +1,310 @@
+﻿using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Data;
+using Avalonia.Input;
+using Avalonia.Input.Platform;
+using Avalonia.Styling;
+using SkiaSharp;
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using FontFamily = Avalonia.Media.FontFamily;
+
+namespace WalletWasabi.Gui.Controls
+{
+	public class NoparaPasswordBox : ExtendedTextBox, IStyleable
+	{
+		Type IStyleable.StyleKey => typeof(TextBox);
+
+		public static readonly string[] Titles =
+		{
+			"这个笨老外不知道自己在写什么。", // This silly foreigner does not know what he is writing.
+			"法式炸薯条法式炸薯条法式炸薯条", //French fries, french fries, french fries
+			"只有一支筷子的人会挨饿。", //Man with one chopstick go hungry.
+			"说太多灯泡笑话的人，很快就会心力交瘁。", // Man who tell one too many light bulb jokes soon burn out!
+			"汤面火锅", //Noodle soup, hot pot
+			"你是我见过的最可爱的僵尸。", //You’re the cutest zombie I’ve ever seen.
+			"永不放弃。", //Never don't give up.
+			"如果你是只宠物小精灵，我就选你。" //If you were a Pokemon, I'd choose you.
+		};
+
+		//public static readonly string[] Titles =
+		//{
+		//	"apple",
+		//	"banana",
+		//	"grape",
+		//	"orange",
+		//	"blackberry",
+		//};
+
+		private static readonly Key[] SuppressedKeys =
+			{ Key.LeftCtrl, Key.RightCtrl, Key.LeftAlt, Key.RightAlt, Key.LeftShift, Key.RightShift, Key.Escape, Key.CapsLock, Key.NumLock, Key.LWin, Key.RWin };
+
+		private bool _supressChanges;
+		private string _displayText = "";
+		private Random _random = new Random();
+		private StringBuilder _sb = new StringBuilder();
+		private HashSet<Key> _supressedKeys;
+
+		public int SelectionLength => SelectionEnd - SelectionStart;
```

Comment:
I set the selection in the control programatically. You can have all or none selected to avoid selecting and removing something from the middle of the text.

Reply:
Added Abs just to make it sure.